Tella/Data/Database/TellaData.swift

```diff
@@ -125,6 +125,23 @@ class TellaData : ObservableObject {
         return database.getReport(reportId: reportId)
     }
     
+    func getResources() -> [DownloadedResource] {
+        return database.getDownloadedResources()
+    }
+    
+    func addResource(resource: Resource, serverId: Int) throws -> String {
```

Comment:
I think it's better to return the `Result` type : `Result<String, Error>` and then treat the result

Reply:
Do you still think we should return Result<String, Error> after moving all the logic to tellaData? After moving the vaultManager to tellaData I'm not returning anything because I save the file in the success handler of the operation. Same with the delete one

Reply:
bringing this back @dhekra-rouatbi. Since we remove all the logic to tella data do we still need to return a Result type?

Reply:
What do you think of this ?

    func addResource(resource: Resource, serverId: Int, data: Data) -> Bool {
        let result = database.addDownloadedResource(resource: resource, serverId: serverId)
        
        switch result {
        case .success(let resourceId):
            guard let isSaved = self.vaultManager?.save(data, vaultFileId: resourceId) else {
                return false
            }
            return true
        case .failure :
            return false
        }
    }